Dockerfile

```diff
@@ -0,0 +1,54 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+#
+# This is part of the project Firefox Profiler.
+#
+# This file assumes that the current directory is in a known state.
+# Especially it's not meant to be run directly, but rather using the command
+# `yarn docker`.
+# Then the command `yarn docker:run` will conveniently execute the image. It's
+# possible to pass extra arguments to this command to execute another program
+# instead of the default.
+
+# Build project
+FROM node:10-buster-slim AS builder
+ENV NODE_ENV="production"
+ENV PORT=8000
+EXPOSE ${PORT}
+
+# We create 2 different users as a "security in depth" measure:
+# - 'installer' is used to install the application.
+# - 'app' is used to run the application and should have read-only access to the
+#   app files.
+RUN set -x \
+    && groupadd --gid 10001 app \
+    && useradd -M --uid 10001 --gid 10001 app
+RUN set -x \
+    && groupadd --gid 10002 installer \
+    && useradd -M --uid 10002 --gid 10002 installer
+RUN mkdir /app
+RUN chown installer:installer /app
+RUN apt-get update && apt-get upgrade -y
+
+# Copy the babelified source as well as the package manager's configuration
+# files. They're owned by the 'installer' user.
+COPY --chown=installer:installer package.json yarn.lock /app/
+COPY --chown=installer:installer dist /app/dist/
```

Comment:
The main reason is because the code that serves `version.json` looks it up in `../dist/version.json` so that it works in the normal dev environment too. So that's why I keep the same directory tree in the container.

Reply:
hmm, it seems weird to have a mention of `dist` in the app code no?

Reply:
Yeah I'm not so comfortable with it either, especially that then it's a bit sensitive to the setup. I think we can revisit, I'll file a separate issue about that.